internal/sync.go

```diff
@@ -643,8 +646,9 @@ func (s *syncGSuite) getGoogleDirectory() ([]*admin.Group, []*admin.User, map[st
                 	}
 
 			// Retrieve the user and add to list of users to sync
-                	uniqueUsers[m.Email] = s.getGoogleUser(m.Email)
-			uniqueMembers[m.Email] = s.getGoogleUser(m.Email)
```

Comment:
This two-time lookup of the user was causing a huge time sync.  Caching the user saved us over a minute of time for each run of the sync.